### PR TITLE
feat: migrate password settings to mantine

### PR DESCRIPTION
**Pull request by joaoviana**
Closes: #5315

### Description:

before
image

after
image

### CODE REVIEW

File: packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx [hunk 1]
```diff
@@ -46,60 +52,48 @@ const PasswordPanel: FC = () => {
...
-                onClick={handleUpdate}
-                loading={isLoading}
-            />
-        </div>
+
+                <Button
+                    mt="md"
+                    type="submit"
+                    ml="auto"
+                    display="flex"
```
Comment (IrakliJani) [inline packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx:87]: no need for flex

File: packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx [hunk 2]
```diff
@@ -46,60 +52,48 @@ const PasswordPanel: FC = () => {
...
-                newPassword,
-            });
-        } else {
-            showToastError({
-                title: 'Required fields: password and new password',
-                timeout: 3000,
-            });
-        }
-    };
+    const onSubmit = form.onSubmit(({ currentPassword, newPassword }) => {
```
Comment (IrakliJani) [inline packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx:55]: I know this is an old code.
im not a big fan of useEffect used like that above...
can you remove useEffect and add `onError` callback to the hook?

File: packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx (hunk 1 above)
Comment (IrakliJani) [inline packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx:87]: you can use Stack wrapper to avoid `mt="md"` props individually

Comment (IrakliJani) [reply packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx:87]: or you can add `gap` to the wrapper

            <Box
                component="form"
                display='flex'
                dir='column'
                h='fit-content'
                sx={{ gap: 'md' }}

File: packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx [hunk 3]
```diff
@@ -46,60 +52,48 @@ const PasswordPanel: FC = () => {
...
-            mutate({
-                password: '',
-                newPassword,
-            });
-        } else {
-            showToastError({
-                title: 'Required fields: password and new password',
-                timeout: 3000,
-            });
-        }
```
[the remark this answers is not in the document]
Comment (IrakliJani) [reply packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx:65]: I am not sure.. can you replicate the logic? user has no password if user signs up with oauth

**Review by IrakliJani — CHANGES_REQUESTED** [review]

File: packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx (hunk 1 above)
Comment (joaoviana) [reply packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx:87]: incredible... this is what I needed!!

Comment (joaoviana) [reply packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx:87]: changed to `display='block'` similar to comment on: https://github.com/lightdash/lightdash/pull/5326#discussion_r1183341274

File: packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx (hunk 2 above)
Comment (joaoviana) [reply packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx:55]: sure, checking now

File: packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx [hunk 4]
```diff
@@ -46,60 +52,48 @@ const PasswordPanel: FC = () => {
...
     return (
-        <div
-            style={{
-                height: 'fit-content',
-                display: 'flex',
-                flexDirection: 'column',
-            }}
-        >
-            {hasPassword && (
+        <Flex dir="column" sx={{ height: 'fit-content' }}>
```
[the remark this answers is not in the document]
Comment (joaoviana) [reply packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx]: Thanks! I actually just checked and we don't need the Flex at all! 
just `form` is sufficient 🎉

File: packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx (hunk 1 above)
Comment (IrakliJani) [reply packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx]: nice! you can use same container here as well: https://github.com/lightdash/lightdash/pull/5326/files

Comment (joaoviana) [reply packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx]: doing now!

Comment (joaoviana) [reply packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx]: done @IrakliJani

File: packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx [hunk 5]
```diff
@@ -34,72 +36,55 @@ const PasswordPanel: FC = () => {
...
+            (error) => {
+                const [title, ...rest] = error.error.message.split('\n');
+                showError({
+                    title,
+                    body: rest.join('\n'),
+                });
+            },
+            [showError],
+        ),
     });
```
Comment (IrakliJani) [inline packages/frontend/src/components/UserSettings/PasswordPanel/index.tsx:49]: nice!

**Review by IrakliJani — APPROVED** [review]
🚢 !

**Pull request merged**